gramex/handlers/mlhandler.py

```diff
@@ -261,14 +271,56 @@ def _transform(self, data, **kwargs):
         data = self._filterrows(data, **kwargs)
         return data
 
+    def _chooseScore(self, data, target):
+        try:
+            print( 'What score are we reading here? -----> ', self.score_pref)
+            print('What class category are we reading here? -----> ', self.multiclass)
+            predict = self.model.predict(data)
+            print('PREDICT ----->', predict)
+            print('TARGET ------>', target)
+
+            #FOR CLASSIFICATION
+            if(self.score_pref == 'f1-score'):
+                return f1_score(target, predict, average= 'weighted') # can change average value
+
+            elif(self.score_pref == 'accuracy'):
+                return accuracy_score(target, predict)
+
+
+            elif(self.score_pref == 'recall'):
+                return recall_score(target, predict, average=None)
+
+            elif(self.score_pref == 'roc_auc_score'):
+                result = roc_auc_score(target, predict)
+                if(self.multiclass == 'True'):
+                    return roc_auc_score(target, predict, multi_class='ovr')
+                return result 
+
+            #FOR REGRESSION    
+            #elif(self.score_pref == 'mean_squared_error'):
+            #   return mean_squared_error(target, predict)
+
+            elif(self.score_pref == 'r2'):
+                return r2_score(target, predict)
+
+            else:
+                return self.model.score(data,target)
+
+        except ValueError:
+            print('ValueError occured. Please check if the scoring metric is appropriate for the given data!')
+
+
+
```

Comment:
Please remove all the print statements here.

Reply:
Also, the way this function is written, we are restricting the user to choosing between the scoring methods imported at the top of the file.

The intention is to let the user select **any** scoring method, and if it's not applicable, raise an error. We can take any string specified by the user, and make a scorer function out of it: [see this function](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.get_scorer.html). It can be used as follows:

```python
from sklearn.metrics import get_scorer
user_defined_metric = 'accuracy'  # or roc_auc_score, r2, etc.
scorer = get_scorer(user_defined_metric)
accuracy = scorer(model, data, target)
```